smart-contracts/contracts/cl-vault/src/msg.rs

```diff
@@ -1,32 +1,51 @@
 use cosmwasm_schema::{cw_serde, QueryResponses};
-use cosmwasm_std::{Decimal, Uint128};
+use cosmwasm_std::{Addr, Decimal, Uint128};
 use cw_dex_router::operations::SwapOperationsListUnchecked;
 use cw_vault_multi_standard::{VaultStandardExecuteMsg, VaultStandardQueryMsg};
 
 use crate::{
     query::{
         AssetsBalanceResponse, PoolResponse, PositionResponse, RangeAdminResponse,
-        UserRewardsResponse, UserSharesBalanceResponse, VerifyTickCacheResponse,
+        UserSharesBalanceResponse, VerifyTickCacheResponse,
     },
     state::{Metadata, VaultConfig},
+    vault::autocompound::SwapAsset,
 };
 
 /// Extension execute messages for an apollo autocompounding vault
 #[cw_serde]
 pub enum ExtensionExecuteMsg {
     /// Execute Admin operations.
     Admin(AdminExtensionExecuteMsg),
+    /// An interface of certain vault interaction with forced values for authz
+    Authz(AuthzExtension),
     /// Rebalance our liquidity range based on an off-chain message
     /// given to us by RANGE_ADMIN
     ModifyRange(ModifyRangeMsg),
     /// provides a fungify callback interface for the contract to use
     Merge(MergePositionMsg),
-    /// Collect any rewards from Osmosis to the Vault
-    CollectRewards { amount_of_users: Uint128 },
+    /// provides an entry point for autocompounding idle funds to current position
+    Autocompound {},
     /// Distribute any rewards over all users
-    DistributeRewards { amount_of_users: Uint128 },
-    /// Claim rewards belonging to a single user
-    ClaimRewards {},
+    CollectRewards {},
+    /// Build tick exponent cache
+    BuildTickCache {},
```

Comment:
This entrypoint is exposed twice, here and in vault::admin.rs as `AdminExtensionExecuteMsg`, we should remove one.